File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int totalSum=0;
        int result=-1;
        for(int num:nums){
            totalSum+=num;
        }
        int rightSum=0;
        for(int i=nums.size()-1;i>-1;i--){
            totalSum-=nums[i];
            if(totalSum==rightSum){
                result=i;
            }
            rightSum+=nums[i];
        }
        return result;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        dpL = [0 for num in nums]
        dpR = [0 for num in nums]
        runningSum = 0 
        for index,num in enumerate(nums):
            dpL[index] = runningSum
            runningSum+=num
        runningSum = 0
        for i in range(len(nums)-1, -1, -1):
            dpR[i] = runningSum
            runningSum+=nums[i]
        for i in range(len(nums)):
            if dpL[i]==dpR[i]:
                return i
        return -1